src/tests/ftest/launch.py

```diff
@@ -465,16 +465,20 @@ class TestInfo():
         "client_users",
     ]
 
-    def __init__(self, test_file, order):
+    def __init__(self, test_file, order, yaml_extension=None):
         """Initialize a TestInfo object.
 
         Args:
             test_file (str): the test python file
             order (int): order in which this test is executed
+            yaml_extension (str, optional): if defined and a test yaml file exists with this
+                extension, the yaml file will be used in place of the default test yaml file.
         """
         self.name = TestName(test_file, order, 0)
         self.test_file = test_file
         self.yaml_file = ".".join([os.path.splitext(self.test_file)[0], "yaml"])
+        if yaml_extension and os.path.exists(".".join([self.yaml_file, str(yaml_extension)])):
+            self.yaml_file = ".".join([self.yaml_file, str(yaml_extension)])
```

Comment:
So we might have an extension like:
```
my_test.yaml.my_ext
```
Do we want to support something more generic like this?
```
my_test.my_ext.yaml
```
The main reason being that linters, etc. won't recognize "yaml.my_ext" as a yaml file

Reply:
Done.